dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+
+        WebClientProvider provider = getWebClientProvider(clientConfiguration);
+
+        HostProvider hostProvider = HostProvider.provider(provider,
+                clientConfiguration.getEndpoints().toArray(new InetSocketAddress[0]));
+        return new DotReactiveElasticsearchClient(hostProvider);
+    }
+
+    private static WebClientProvider getWebClientProvider(ClientConfiguration clientConfiguration) {
+
+        Duration connectTimeout = clientConfiguration.getConnectTimeout();
+        Duration soTimeout = clientConfiguration.getSocketTimeout();
+
+        TcpClient tcpClient = TcpClient.create();
+
+        if (!connectTimeout.isNegative()) {
+            tcpClient = tcpClient.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, Math.toIntExact(connectTimeout.toMillis()));
+        }
+
+        if (!soTimeout.isNegative()) {
+
+            tcpClient = tcpClient.doOnConnected(connection -> connection //
+                    .addHandlerLast(new ReadTimeoutHandler(soTimeout.toMillis(), TimeUnit.MILLISECONDS))
+                    .addHandlerLast(new WriteTimeoutHandler(soTimeout.toMillis(), TimeUnit.MILLISECONDS)));
+        }
+
+        String scheme = "http";
+        HttpClient httpClient = HttpClient.from(tcpClient);
+
+        if (clientConfiguration.useSsl()) {
+
+            httpClient = httpClient.secure(sslConfig -> {
+
+                Optional<SSLContext> sslContext = clientConfiguration.getSslContext();
+                sslContext.ifPresent(it -> sslConfig.sslContext(new JdkSslContext(it, true, ClientAuth.NONE)));
+            });
+
+            scheme = "https";
+        }
+
+        ReactorClientHttpConnector connector = new ReactorClientHttpConnector(httpClient);
+        WebClientProvider provider = WebClientProvider.create(scheme, connector);
+
+        if (clientConfiguration.getPathPrefix() != null) {
+            provider = provider.withPathPrefix(clientConfiguration.getPathPrefix());
+        }
+
+        provider = provider.withDefaultHeaders(clientConfiguration.getDefaultHeaders()) //
+                .withWebClientConfigurer(clientConfiguration.getWebClientConfigurer());
+        return provider;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders)
+     */
+    @Override
+    public Mono<Boolean> ping(HttpHeaders headers) {
+
+        return sendRequest(new MainRequest(), DotReactiveElasticsearchClient.RequestCreator.ping(), RawActionResponse.class, headers) //
+                .map(response -> response.statusCode().is2xxSuccessful()) //
+                .onErrorResume(NoReachableHostException.class, error -> Mono.just(false)).next();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#info(org.springframework.http.HttpHeaders)
+     */
+    @Override
+    public Mono<MainResponse> info(HttpHeaders headers) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'headers' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)